### PR TITLE
Handle mouse down event on scrollbar

### PR DESCRIPTION
**Pull request by jofuria**
Handle mouse down event.
The scroll bar will scroll to the position the user has clicked.

### CODE REVIEW

File: src/web/CustomScrollbar.ts [hunk 1]
```diff
@@ -329,7 +329,24 @@ export class Scrollbar {
     }
 
     private _handleMouseDown(e: React.MouseEvent<any>) {
-        this._prevent(e);
+        const target = e.currentTarget as any;
+
+        if (this._dragging === true || e.target === this._verticalBar.slider || !target) {
```
Comment (erictraut) [inline src/web/CustomScrollbar.ts]: No need to compare against "true" because "_dragging" is already a boolean.

File: src/web/CustomScrollbar.ts [hunk 2]
```diff
@@ -329,7 +329,24 @@ export class Scrollbar {
...
+            this._prevent(e);
+            return;
+        }
+
+        if (this._hasVertical) {
+            const eventOffsetY = e.pageY - target.getBoundingClientRect().top;
+            const halfHeight = this._verticalBar.slider.offsetHeight / 2; 
+            const offsetY = (eventOffsetY - this._verticalBar.slider.offsetTop - halfHeight) * this._verticalBar.slider2Scroll;
+            this._viewport.scrollTop = offsetY + this._viewport.scrollTop;
+        } else if (this._hasHorizontal) {
```
Comment (erictraut) [inline src/web/CustomScrollbar.ts]: What if it has both a horizontal and vertical? I think the "else" should be removed.

File: src/web/CustomScrollbar.ts [hunk 3]
```diff
@@ -329,7 +329,24 @@ export class Scrollbar {
...
+            const eventOffsetX = e.pageX - target.getBoundingClientRect().left;
+            const halfWidth = this._horizontalBar.slider.offsetWidth / 2; 
+            const offsetX = (eventOffsetX - this._horizontalBar.slider.offsetLeft - halfWidth) * this._verticalBar.slider2Scroll;
```
Comment (erictraut) [inline src/web/CustomScrollbar.ts]: This should use this._horizontalBar, not this._verticalBar.

Comment (jofuria) [reply src/web/CustomScrollbar.ts]: argh old version

**Pull request closed without merging**